### ISSUE TITLE
Add a Tidal Analysis vignette

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2243]
This came up in an email, and I'm making an issue so the work can be cross-referenced.

**Comment by dankelley** [comment]
@richardsc I just pushed (with new vignette) to GH, including the docs, so the webpage has a semi-blank entry for tides.  I won't likely get time to put in the "univariate case" -- the `tidem()` case -- until next week.  I don't think we will get merge conflicts, if you only edit in the "bivariate case" block and I only edit in the "univariate case" block.  But even if we get conflicts, it's not like it will break code.

The best way to check the building of the vignette is to do that in RStudio.  Please don't use `pkgdown::build_site()` because I think that inserts a date in every single html file (one per function, plus for vignettes and datasets) and your date and my date will conflict so every single file will have a merge conflict.

**Comment by richardsc** [comment]
I pushed some material, mostly based on what was on my blog post, but it's causing some crashes of the auto-build (not surprising, given the external dependencies required). Perhaps we can chat this week about how best to write/include a vignette that we don't actually want the system (e.g. CRAN) to build.

**Comment by dankelley** [comment]
Sure.  Basically the answer is to enclose code in triple backticks.  That won't colourize it, but I think "triple-backtick R" (that is, with no braces) might do that.  For figures, they need to be included like normal markdown.  We need to be tight on diagrams space so the fewer the better, and ideally use preview and save to jpg with enough compression to get say 100K or maybe 150K.

**Comment by richardsc** [comment]
Ok, so the idea is to create the output (figures etc) separately and then include it directly?

**Comment by dankelley** [comment]
Yup.  Basically, don't run any code, unless the CRAN system will have all the software (which it won't ... no python, for example).

**Comment by richardsc** [comment]
👍

**Comment by dankelley** [comment]
I wanted to clear this off the to-do list, so I took some tide-related things from the main vignette and put them here. That includes 2 exercises.  Obviously, we can make improvements later but this is enough to close the present issue.  (The webpage has the vignette, in case anyone wants to take a look.  I did not make figures show, out of consideration for package size. Besides, users ought to copy/paste/run code instead of passive looking at images.)

commit cca779c892d915f1402918ca98efda9457ab2e27
Author: dankelley <kelley.dan@gmail.com>
Date:   Sat Dec 7 14:26:24 2024 -0400

    add tidal vignette
